song_translator/track/views.py

```diff
@@ -68,14 +74,16 @@ def get(self, request, pk):
 
     def post(self, request, pk):
         translation_data = JSONParser().parse(request)
-        if translation_data["auto_translate"]:
-            translator = google_translator()
-            track = Track.objects.get(id=pk)
-            track_serializer = TrackSerializer(track)
-            translation_data["text"] = translator.translate(track_serializer.data['text'],
-                                                            lang_tgt=translation_data["language"],
-                                                            lang_src=track_serializer.data[
-                                                                'original_language'])
+        if get_auto_translate_flag():
+            if translation_data["auto_translate"]:
+                translator = google_translator()
+                track = Track.objects.get(id=pk)
+                track_serializer = TrackSerializer(track)
+                translation_data["text"] = translator.translate(track_serializer.data['text'],
+                                                                lang_tgt=translation_data["language"],
+                                                                lang_src=track_serializer.data[
+                                                                    'original_language'])
+        ldclient_close()
```

Comment:
better place to call it is when request handling ends. I think this can be achieved with signals https://docs.djangoproject.com/en/3.2/ref/signals/#django.core.signals.request_finished. Also middleware can be a way to go.